galoisAvx512_amd64.go

```diff
@@ -0,0 +1,181 @@
+//+build !noasm
+//+build !appengine
+//+build !gccgo
+
+// Copyright 2015, Klaus Post, see LICENSE for details.
+// Copyright 2019, Minio, Inc.
+
+package reedsolomon
+
+//go:noescape
+func _galMulAVX512Parallel82(in, out [][]byte, matrix *[(16 + 16) * dimIn * dimOut82]byte, addTo bool)
+
+//go:noescape
+func _galMulAVX512Parallel84(in, out [][]byte, matrix *[(16 + 16) * dimIn * dimOut84]byte, addTo bool)
+
+const dimIn = 8
+const dimOut82 = 2
+const dimOut84 = 4
+
+var zeroCoeff = [16]byte{}
+
+// Construct block of matrix coefficients for 2 outputs rows in parallel
+func setupMatrix82(matrixRows [][]byte, inputOffset, outputOffset int, matrix *[(16 + 16) * dimIn * dimOut82]byte) {
+
+	offset := 0
+	for c := inputOffset; c < inputOffset+dimIn; c++ {
+		for iRow := outputOffset; iRow < outputOffset+dimOut82; iRow++ {
+			if c < len(matrixRows[iRow]) {
+				coeff := matrixRows[iRow][c]
+				copy(matrix[offset*32:], mulTableLow[coeff][:])
+				copy(matrix[offset*32+16:], mulTableHigh[coeff][:])
+			} else {
+				// coefficients not used for this input shard (so null out)
+				copy(matrix[offset*32:], zeroCoeff[:])
```

Comment:
Usually I would expect this to be faster:

```Go
v := matrix[offset*32:offset*32+32]
for i := range v {
   v[i] = 0
}
```

The go compiler should be able to do a pure duffzero/memset.

Not sure it has practical implications though.

Reply:
Good remark, will change and eliminate zeroCoeff slice.